interfaces/builtin/raw_volume_test.go

```diff
@@ -0,0 +1,333 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type rawVolumeInterfaceSuite struct {
+	testutil.BaseTest
+	iface interfaces.Interface
+
+	// OS snap
+	testSlot1     *interfaces.ConnectedSlot
+	testSlot1Info *snap.SlotInfo
+
+	// Gadget snap
+	testUDev1     *interfaces.ConnectedSlot
+	testUDev1Info *snap.SlotInfo
+	testUDev2     *interfaces.ConnectedSlot
+	testUDev2Info *snap.SlotInfo
+	testUDev3     *interfaces.ConnectedSlot
+	testUDev3Info *snap.SlotInfo
+
+	testUDevBadValue1     *interfaces.ConnectedSlot
+	testUDevBadValue1Info *snap.SlotInfo
+
+	// Consuming snap
+	testPlugPart1     *interfaces.ConnectedPlug
+	testPlugPart1Info *snap.PlugInfo
+	testPlugPart2     *interfaces.ConnectedPlug
+	testPlugPart2Info *snap.PlugInfo
+	testPlugPart3     *interfaces.ConnectedPlug
+	testPlugPart3Info *snap.PlugInfo
+}
+
+var _ = Suite(&rawVolumeInterfaceSuite{
+	iface: builtin.MustInterface("raw-volume"),
+})
+
+const rawVolumeConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [raw-volume]
+`
+
+const rawVolumeCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  raw-volume:
+`
+
+func (s *rawVolumeInterfaceSuite) SetUpTest(c *C) {
+	// Mock for OS snap
+	osSnapInfo := snaptest.MockInfo(c, `
+name: core
+version: 0
+type: os
+slots:
+  test-part-1:
+    interface: raw-volume
+    path: /dev/vda1
+  test-part-2:
+    interface: raw-volume
+    path: /dev/mmcblk0p1
+  test-part-3:
+    interface: raw-volume
+    path: /dev/i2o/hda1
+`, nil)
+	s.testSlot1Info = osSnapInfo.Slots["test-part-1"]
+	s.testSlot1Info = osSnapInfo.Slots["test-part-2"]
+	s.testSlot1Info = osSnapInfo.Slots["test-part-3"]
```

Comment:
what's the purpose of re-assigning `s.testSlot1Info` again like this? Feels like you meant to have 3 variables `s.testSlot1Info`, `s.testSlot2Info`, and `s.testSlot3Info` for this, but there's only `testSlot1Info` in the struct